src/RouterInterface.php

```diff
@@ -57,13 +54,9 @@ public function match(Request $request);
      * the URI, this should be performed afterwards; consider passing the URI
      * to league/uri to encode it.
      *
-     * @see https://github.com/auraphp/Aura.Router#generating-a-route-path
-     * @see http://framework.zend.com/manual/current/en/modules/zend.mvc.routing.html
-     * @param string $name
-     * @param array $substitutions
-     * @param array $options
-     * @return string
+     * @see https://github.com/auraphp/Aura.Router/blob/3.x/docs/generating-paths.md
+     * @see https://docs.zendframework.com/zend-router/routing/
      * @throws Exception\RuntimeException if unable to generate the given URI.
      */
-    public function generateUri($name, array $substitutions = [], array $options = []);
+    public function generateUri(string $name, array $substitutions = [], array $options = []) : string;
```

Comment:
And this one **requires** a major version bump, as the typehint of an argument has changed.

I think we should limit the changes in this interface to return value typehints for now, as it will result in fewer moving parts with regards to coordinating releases.

We can revisit after we have all the various repos updated.